packages/devtools_server/lib/src/handlers.dart

```diff
@@ -103,3 +177,182 @@ class ServerApi {
+  }
+
+  /// Does the .flutter store exist?
+  static bool get doesStoreExits {
+    final flutterStore = File('${DevToolsUsage.userHomeDir()}/.flutter');
+    return flutterStore.existsSync();
+  }
+
+  bool get isFirstRun => _analytics.firstRun;
+
+  bool get enabled => _analytics.enabled;
+
+  set enabled(bool value) => _analytics.enabled = value;
+
+  String get clientId => _analytics.clientId;
+}
+
+// Access the DevTools on disk store (~/.devtools).
+class DevToolsUsage {
+  /// Create a new Usage instance; [versionOverride] and [configDirOverride] are
+  /// used for testing.
+  DevToolsUsage(
+      {String settingsName = 'devtools',
+      String versionOverride,
+      String configDirOverride}) {
+    // final FlutterVersion flutterVersion = FlutterVersion.instance;
+    // final String version = versionOverride ?? flutterVersion.getVersionString(redactUnknownBranches: true);
+    // TODO(terry): UA, first parameter, is '' could be DevTools UA
+    // TODO(terry): version, second parameter, is '' could be real Flutter version #.
+    // TODO(terry): documentDirectory, third parameter, is null could be :
+    //    documentDirectory: configDirOverride != null ? fs.directory(configDirOverride) : null
+
+    properties =
+        IOPersistentProperties(settingsName, documentDirPath: userHomeDir());
+  }
+
+  static String userHomeDir() {
+    final String envKey =
+        Platform.operatingSystem == 'windows' ? 'APPDATA' : 'HOME';
+    final String value = Platform.environment[envKey];
+    return value == null ? '.' : value;
+  }
+
+  IOPersistentProperties properties;
+
+  void reset() {
+    properties.remove('firstRun');
+    properties['enabled'] = false;
+    properties['surveyShownCount'] = 0;
+    properties['surveyActionTaken'] = false;
+  }
+
+  bool get isFirstRun {
+    properties['firstRun'] = properties['firstRun'] == null;
+    return properties['firstRun'];
+  }
+
+  bool get enabled {
+    if (properties['enabled'] == null) {
+      properties['enabled'] = false;
+    }
+
+    return properties['enabled'];
+  }
+
+  set enabled(bool value) {
+    properties['enabled'] = value;
+    return properties['enabled'];
+  }
+
+  int get surveyShownCount {
+    if (properties['surveyShownCount'] == null) {
+      properties['surveyShownCount'] = 0;
+    }
+
+    return properties['surveyShownCount'];
+  }
+
+  void incrementSurveyShownCount() {
+    surveyShownCount; // Insure surveyShownCount has been initialized.
+    properties['surveyShownCount'] += 1;
+  }
+
+  bool get surveyActionTaken => properties['surveyActionTaken'] == true;
+
+  set surveyActionTaken(bool value) {
+    properties['surveyActionTaken'] = value;
+  }
+}
+
+abstract class PersistentProperties {
+  PersistentProperties(this.name);
+
+  final String name;
+
+  dynamic operator [](String key);
+  void operator []=(String key, dynamic value);
+
+  /// Re-read settings from the backing store. This may be a no-op on some
```

Comment:
nit: blank line after first sentence of dart doc

Reply:
Done.